scipy/signal/_splinemodule.cc

```diff
@@ -0,0 +1,513 @@
+
+    Py_DECREF(a_image);
+    Py_DECREF(a_hrow);
+    Py_DECREF(a_hcol);
+    return PyArray_Return(out);
+
+fail:
+    Py_XDECREF(a_image);
+    Py_XDECREF(a_hrow);
+    Py_XDECREF(a_hcol);
+    Py_XDECREF(out);
+    return NULL;
+}
+
+static char doc_IIRsymorder1_ic[] = "out = symiirorder1_ic(input, z1, precision=-1.0)\n"
+                                    "\n"
+                                    "    Compute the (forward) mirror-symmetric boundary conditions for a smoothing\n"
+                                    "    IIR filter that is composed of cascaded first-order sections.\n"
+                                    "\n"
+                                    "    The starting condition returned by this function is computed based on\n"
+                                    "    the following transfer function::\n"
+                                    "\n"
+                                    "                       1         \n"
+                                    "           H(z) = ------------   \n"
+                                    "                   (1 - z1/z)    \n"
+                                    "\n"
+                                    "\n"
+                                    "    Parameters\n"
+                                    "    ----------\n"
+                                    "    input : ndarray\n"
+                                    "        The input signal. If 2D, then it will find the initial conditions \n"
+                                    "        for each of the elements on the last axis.\n"
+                                    "    z1 : scalar\n"
+                                    "        Parameter in the transfer function.\n"
+                                    "    precision :\n"
+                                    "        Specifies the precision for calculating initial conditions\n"
+                                    "        of the recursive filter based on mirror-symmetric input.\n"
+                                    "\n"
+                                    "    Returns\n"
+                                    "    -------\n"
+                                    "    z_0 : ndarray\n"
+                                    "        The mirror-symmetric initial condition for the forward IIR filter.";
+
+static PyObject *IIRsymorder1_ic(PyObject *NPY_UNUSED(dummy), PyObject *args) {
+    PyObject *sig = NULL;
+    PyArrayObject *a_sig = NULL, *out = NULL;
+    npy_intp *in_size;
+
+    Py_complex z1;
+    double precision = -1.0;
+    int thetype, ret;
+    npy_intp M, N;
+    PyArray_Descr *dtype;
+
+    if (!PyArg_ParseTuple(args, "OD|d", &sig, &z1, &precision))
+        return NULL;
+
+    thetype = PyArray_ObjectType(sig, NPY_FLOAT);
+    thetype = PyArray_MIN(thetype, NPY_CDOUBLE);
+    a_sig = (PyArrayObject *) PyArray_FromObject(sig, thetype, 1, 2);
+
+    if (a_sig == NULL) {
+        // goto fail: block
+        Py_XDECREF(a_sig);
+        Py_XDECREF(out);
+        return NULL;
```

Comment:
There is a `fail:` label in this function, so why are we duplicating this? Either spell it out everywhere and get rid of the `fail:` label, or keep the label and `goto fail` wherever required.

Reply:
Ah, a C++ fun. `goto fail;`  from here crosses an initialization statement a few lines below, and fails to compile with a "crosses initialization" error.
I guess a comment needs expanding.